### PR TITLE
fix(inventory): check if link need to be updated are added (prevent d…

### PR DESCRIPTION
**Pull request by stonebuzz**
At each inventory, GLPI delete all links (and ```infocom``` too) between asset and device (```item_devicexxxx```)

Check if link need to be updated are added (prevent bulk deletion)

Best regards

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #12852

### CODE REVIEW

**Review by orthagh — APPROVED** [review]
Only concern on my side, Did we still have some good performance with a SELECT+(UPDATE|INSERT) in a foreach

**Comment by stonebuzz** [comment]
@orthagh 

Before this fix : 
```shell
Computer:
	prepare:
		Execution time:      0.0003 seconds
		Memory usage:        5.69 Mio
		Memory usage (real): 10 Mio
		Memory peak:         8.99 Mio

	handle:
		Execution time:      2.12774 seconds
		Memory usage:        10.36 Mio
		Memory usage (real): 18 Mio
		Memory peak:         13.23 Mio

	full:
		Execution time:      3.46368 seconds
		Memory usage:        10.35 Mio
		Memory usage (real): 18 Mio
		Memory peak:         13.23 Mio

Inventoried assets: Computer #1
```

After : 

```
Computer:
	prepare:
		Execution time:      0.00029 seconds
		Memory usage:        5.69 Mio
		Memory usage (real): 10 Mio
		Memory peak:         8.99 Mio

	handle:
		Execution time:      2.06414 seconds
		Memory usage:        10.4 Mio
		Memory usage (real): 18 Mio
		Memory peak:         13.28 Mio

	full:
		Execution time:      3.30758 seconds
		Memory usage:        10.39 Mio
		Memory usage (real): 18 Mio
		Memory peak:         13.28 Mio

Inventoried assets: Computer #1
```

File: src/Inventory/Asset/Device.php [hunk 1]
```diff
@@ -102,15 +102,27 @@ public function handle()
...
                 $input_data = \Toolbox::addslashes_deep([
                     $fk                  => $device_id,
                     'itemtype'           => $this->item->getType(),
                     'items_id'           => $this->item->fields['id'],
+                    'is_dynamic'         => 1,
                 ]);
 
+                //add serial if available (usefull for memories)
+                if (property_exists($val, "serial")) {
+                    $input_data['serial'] = $val->serial;
```
Comment (trasher) [inline src/Inventory/Asset/Device.php:118]: Merging whole `$input_device` should cover all cases, no?

**Comment by stonebuzz** [comment]
I close in favour of https://github.com/glpi-project/glpi/pull/12904

**Pull request closed without merging**